connectors/sources/confluence.py

```diff
@@ -258,9 +279,18 @@ async def paginated_api_call(self, url_name, **url_kwargs):
                         self.host_url,
                         links.get("next")[1:],
                     )
+            except ServerDisconnectedError:
+                raise
+            except ClientResponseError:
+                raise
+            except ClientPayloadError as exception:
+                self._logger.error(
+                    f"Payload error occurred while fetching {url_name} from {url}. Exception: {exception}."
+                )
+                break
             except Exception as exception:
-                self._logger.warning(
```

Comment:
Why did this change from a warning to an exception? Exception logging will print out the full stack trace and at ERROR level. But we're not crashing/re-raising here, so is this an error, or do we need to warn the user that there _might_ be an issue?